config/7.0/obdemo-bank/ig/routes/routes-service/10-ob-account-consent.json

```diff
@@ -56,25 +71,11 @@
                 "delegate": {
                   "comment":"resolve access tokens and retrieve metadata about the token. The endpoint typically returns the time until the token expires, the OAuth 2.0 scopes associated with the token, and potentially other information",
                   "name": "token-resolver-1",
-                  "type": "TokenIntrospectionAccessTokenResolver",
+                  "type": "StatelessAccessTokenResolver",
                   "config": {
-                    "endpoint": "https://&{identity.platform.fqdn}/am/oauth2/realms/root/realms/&{am.realm}/introspect",
-                    "providerHandler": {
-                      "type": "Chain",
-                      "config": {
-                        "filters": [
-                          {
-                            "type": "HttpBasicAuthenticationClientFilter",
-                            "config": {
-                              "username": "&{ig.agent.id}",
-                              "passwordSecretId": "ig.agent.password",
-                              "secretsProvider": "SystemAndEnvSecretStore-IAM"
-                            }
-                          }
-                        ],
-                        "handler": "ForgeRockClientHandler"
-                      }
-                    }
+                    "secretsProvider": "SecretsProvider-AmJWK",
+                    "issuer": "https://&{identity.platform.fqdn}/am/oauth2/realms/root/realms/&{am.realm}",
+                    "verificationSecretId": "any.value.in.regex.format"
```

Comment:
can you explain what the verificationSecretId configuration is doing please?

Reply:
The verificationSecretIdvconfigurations can be used to specify a secret ID for the secret used to verify the signature of signed access_tokens. In this case I don't think it's used (I have token over that bit from another route where that filter was already implemented in the past) because our JWT's contain the key id under the "kid" claim in the token header.

The documentation states:

The JWT signature is validated as follows:

- If the JWT contains a kid with a mapped value, for example verification.key.1.:
    - The secrets provider queries the KeyStoreSecretStore for a named secret with the secret ID verification.secret.id and the stable ID verification.key.1.
    - Because the KeyStoreSecretStore contains that mapping, the KeyStoreSecretStore returns a named secret.
    - The StatelessAccessTokenResolver tries to validate the JWT signature with the named secret. If it fails, the token is considered as invalid.


- If the JWT contains a kid with an unmapped value, for example, verification.key.3:

    - The secrets provider queries the KeyStoreSecretStore for a named secret with the secret ID verification.secret.id and the stable ID verification.key.3.
    - Because the KeyStoreSecretStore doesn’t contain that mapping, named secret resolution fails. IG tries valid secret resolution in the same way as when the JWT doesn’t contain a kid.
    - If the JWT doesn’t contain a kid:

        The secrets provider queries the KeyStoreSecretStore for all valid secrets, whose alias is mapped to the secret ID verification.secret.id

Reply:
Looks like the field is mandatory for the filter and we have to keep it